tests/test_migrate.py

```
@@ -0,0 +1,1434 @@
+        db_path = legacy_root / "meshtastic.sqlite"
+        # Create corrupt database
+        db_path.write_bytes(b"corrupt data")
+        new_home = tmp_path / "home"
+        new_home.mkdir()
+
+        from mmrelay.migrate import MigrationError
+
+        with pytest.raises(MigrationError, match="Database verification failed"):
+            migrate_database([legacy_root], new_home, move=False)
+
+    def test_most_recent_database_selected(self, tmp_path, monkeypatch):
```

Ruff (0.14.14)

[warning] Unused method argument: `monkeypatch`

(ARG002)